doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,965 @@
+        /// PackageFamilyName and ProductCode can have multiple values.
+        Windows.Foundation.Collections.IVectorView<String> PackageFamilyNames { get; };
+        Windows.Foundation.Collections.IVectorView<String> ProductCodes { get; };
+
+        /// Gets the app catalog  where this package version is from.
+        AppCatalogReference AppCatalogReference { get; };
+
+        /// DESIGN NOTE:
+        /// GetManifest from IPackageVersion in AppInstallerRepositorySearch is not implemented in V1. That class has 
+        /// a lot of fields and no one requesting it.
+        /// Gets the manifest of this package version.
+        /// virtual Manifest::Manifest GetManifest() = 0;
+    }
+
+    /// IMPLEMENTATION NOTE: PackageVersionKey from AppInstallerRepositorySearch.h
+    /// A key to identify a package version within a package.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass PackageVersionId
+    {
+        /// The app catalog id that this version came from.
+        String AppCatalogId { get; };
+        /// The version.
+        String Version { get; };
+        /// The channel.
+        String Channel { get; };
+    };
+
+    /// IMPLEMENTATION NOTE: IPackage from AppInstallerRepositorySearch.h
+    /// A package, potentially containing information about it's local state and the available versions.
+    [contract(Microsoft.Management.Deployment.WindowsPackageManagerContract, 1)]
+    runtimeclass CatalogPackage
+    {
+        /// IMPLEMENTATION NOTE: This would be new work as Windows Package Manager does not currently create a 
+        /// CatalogPackage object from a manifest.
+        /// DESIGN NOTE: This is proposed as the preferred alternative to having InstallOptions match the winget
+        /// command line which allows you to pass in a manifestPath directly into install. Allowing creation of 
+        /// a CatalogPackage object from a manifest allows the install api to consistently use package objects 
+        /// rather than having a "you can either set the manifest path or use a package but not both" problem.
+        static CatalogPackage TryCreateFromManifestPath(String manifestPath);
+
+        /// IMPLEMENTATION NOTE: PackageProperty fields from AppInstallerRepositorySearch.h
+        /// Gets a property of this package.
+        String Id { get; };
+        String Name { get; };
+
+        /// Gets the installed package information if the package is installed.
+        PackageVersionInfo InstalledVersion{ get; };
+
+        /// Gets all available versions of this package.
+        /// The versions will be returned in sorted, descending order.
+        ///  Ex. { 4, 3, 2, 1 }
+        Windows.Foundation.Collections.IVectorView<PackageVersionId> AvailableVersions { get; };
+
+        /// Gets the latest version of this package.
+        PackageVersionInfo LatestAvailableVersion { get; };

```

Comment:
RECOMMEND: `DefaultInstallVersionInfo` ... this `CatalogPackage` object is a snapshot so there's no concerns with hidden IO